src/vs/code/node/cliProcessMain.ts

```diff
@@ -246,7 +242,10 @@ export function main(argv: ParsedArgs): TPromise<void> {
 			const instantiationService2 = instantiationService.createChild(services);
 			const main = instantiationService2.createInstance(Main);
 
-			return main.run(argv);
+			return (main.run(argv) || TPromise.as(null)).then(() => {
```

Comment:
Might just be cleaner to enforce that `run` always returns a promise than doing this `||`. I think this is the only place where we don't return a promise:

https://github.com/Microsoft/vscode/blob/master/src/vs/code/node/cliProcessMain.ts#L80